etsin_finder/frontend/js/components/qvain/utils/select.js

```diff
@@ -31,3 +33,31 @@ export const onChange = (options, lang, callback, constructFunc) => (selection)
     callback(undefined)
   }
 }
+
+export const getOptions = async (ref, inputValue) => {
+  if (!inputValue) return []
+  const api = refDataApi(ref)
+  const response = await api.get(`_search?size=100&q=*${inputValue}*`)
+  return parseRefResponse(response)
+}
+
+const refDataApi = ref =>
+  axios.create({
+    baseURL: `https://metax.csc.local/es/reference_data/${ref}/`,
+  })
+
+const parseRefResponse = res => {
```

Comment:
Some references don't have translations in all languages, e.g. "Luotola (Luumäki)" has no English translation. Use `ref._source.label.en || ref._source.label.und`.